### DISCUSSION TITLE
Signature field

### DISCUSSION THREAD

**Discussion by Suriyasanjay** [MatthiasValvekens/pyHanko#347] [Q&A]
Signature field is not displying  pdf file ?
```
` from pyhanko.sign.fields import append_signature_field, SigFieldSpec
  from pyhanko.pdf_utils.incremental_writer import IncrementalPdfFileWriter
  from pyhanko.sign.signers import PdfSigner
  from pyhanko.keys import load_cert_from_pemder
  from pyhanko.sign import fields
  from pyhanko.sign.fields import SigSeedValueSpec, SigCertConstraints, SigCertConstraintFlags, SigSeedValFlags
  
  
  
  franchising_ca = load_cert_from_pemder('cert.pem')
  
  sv = SigSeedValueSpec(
      reasons=[
          'I vote in favour of the proposed measure',
          'I vote against the proposed measure',
          'I formally abstain from voting on the proposed measure'
      ],
      cert=SigCertConstraints(
          issuers=[franchising_ca],
          flags=SigCertConstraintFlags.ISSUER
      ),
      digest_methods=['sha256', 'sha384', 'sha512'],
      flags=SigSeedValFlags.REASONS | SigSeedValFlags.DIGEST_METHOD
  )
  sp = SigFieldSpec('Signature7', seed_value_dict=sv,
                    box=(200, 100, 400, 160), on_page=0, visible_sig_settings=fields.VisibleSigSettings(rotate_with_page=True, scale_with_page_zoom=True))
  with open('document.pdf', 'rb+') as doc:
      w = IncrementalPdfFileWriter(doc)
      append_signature_field(w, sp)
      w.write_in_place()`
```
      
      
![Screenshot from 2023-11-24 19-41-27](https://github.com/MatthiasValvekens/pyHanko/assets/86243176/9f61bf3d-d4db-4d21-bc5d-10418211e103)
here is the screen shot where there no signature filed after creating

**Comment by MatthiasValvekens** [comment]
Impossible to say for sure without the actual file, but the vast majority of PDF viewers on Linux don't have built-in signature support and will (at best) render annotation appearance streams only. If your empty signature field has an empty appearance stream (which is the default in pyHanko IIRC), then it won't show up. An actual (visible) signature probably will, though.

**Reply by Suriyasanjay** [reply]
Even in the  chrome it not showed the signature field 

![Screenshot from 2023-11-25 09-50-56](https://github.com/MatthiasValvekens/pyHanko/assets/86243176/63655cf1-813f-45a2-baee-b79b984e0970)

**Reply by ag-gaphp** [reply]
Just FYI, browser PDF readers also do not handle signature fields or esigning. I would try something like [Foxit PDF Reader](https://www.foxit.com/downloads/) or [Adobe Acrobat](https://get.adobe.com/reader/) running in Wine to test and see if your signature fields are there. Adobe being the ultimate test I think since they created PDF **_(just make sure you uncheck the option to install McAfee when you download it!!!!!!!!!)_**
